sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StringOffsetMethods.cs

```diff
@@ -4,41 +4,76 @@ namespace Tests.Diagnostics
 {
     public class StringOffsetMethods
     {
-        public void StringOffsetMethodsCases()
+        public StringOffsetMethods()
+        {
+            "Test".Substring(1).IndexOf('t'); // Noncompliant {{Replace 'IndexOf' with the overload that accepts an offset parameter.}}
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        }
+
+        public int GetIndex =>
+            "Test".Substring(1).IndexOf('t'); // Noncompliant {{Replace 'IndexOf' with the overload that accepts an offset parameter.}}

```

Comment:
as discussed, you can have only one or two tests for the message and issue highlighting, and for the rest test simply with `// Noncompliant`